org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/javaeditor/codemining/CalleeJavaMethodParameterVisitor.java

```diff
@@ -165,7 +165,7 @@ private boolean skipParameterNameCodeMining(String[] parameterNames, List<?> arg
 	}
 
 	private boolean skipParameterNamesCodeMinings(IMethod method) {
-		return method.getNumberOfParameters() <= 1;
+		return false;
```

Comment:
Please use a preference here
```suggestion
		return PreferenceConstants.getPreferenceStore().getBoolean(PreferencesConstants.EDITOR_JAVA_CODEMINING_SHOW_PARAMETER_NAME_SINGLE_ARG) && method.getNumberOfParameters() == 1) || method.getNumberOfParameters() == 0;
```

Reply:
Rather the opposite:

```java
boolean showNamesOfSingleArguments= PreferenceConstants.getPreferenceStore().getBoolean(PreferenceConstants.EDITOR_JAVA_CODEMINING_SHOW_PARAMETER_NAME_SINGLE_ARG);
return !showNamesOfSingleArguments && method.getNumberOfParameters() == 1
		|| method.getNumberOfParameters() == 0;
```

Reply:
if you're into renaming with opposite meanings, I suggest renaming (and reversing conditions) from `skipParameterNamesCodeMinings` to `showParameterNamesCodeMinings() { return number > 1 || PreferenceConstants.getPreferenceStore().getBoolean(PreferencesConstants.EDITOR_JAVA_CODEMINING_SHOW_PARAMETER_NAME_SINGLE_ARG))`. That should make things simpler to understand and maintain.

Reply:
No need to rename the method in this PR, it is fine as it is and it is overloaded so the renaming would have to be done for these 4 methods just so the class stays consistent:

![image](https://github.com/user-attachments/assets/c2727989-72ab-47eb-8bd5-deabaecb9b50)